lib/romeo/auth.ex

```diff
@@ -59,19 +67,19 @@ defmodule Romeo.Auth do
   end
 
   defp authenticate_with("DIGEST-MD5", _conn) do
-    raise "Not implemented"
+    raise "Not implemented, please provide an implementation such as preferred_auth_mechanisms: [{\"DIGEST-MD5\", fn conn -> conn end}]"
```

Comment:
Love these improvements. Let's just update it to explain the auth module interface.